### PR TITLE
Feature.isabelle lexer

### PR DESCRIPTION
**Pull request by Dacit**
Adapted Isabelle pygments lexer with some improvements.

### CODE REVIEW

**Comment by Dacit** [comment]
Travis seems to be stuck - is this expected?

**Comment by Dacit** [comment]
@dblessing can I expect this to get merged anytime soon, or should I find a work-around to patch the lexer in for my needs?

**Comment by dblessing** [comment]
@Dacit I've been unable to do much with Rouge lately. It's my intention to get back to it ASAP, but I can't promise anything. 

In the meantime, Travis has been fixed and if you force push it will kick off a build.

**Comment by Dacit** [comment]
> @Dacit I've been unable to do much with Rouge lately. It's my intention to get back to it ASAP, but I can't promise anything.
> 
> In the meantime, Travis has been fixed and if you force push it will kick off a build.

Thanks for the reply; CI runs through now. Would be awesome if you could merge this (changes are self-contained)!

**Comment by Dacit** [comment]
@dblessing

**Comment by Dacit** [comment]
FYI this is still open

**Comment by tancnle** [comment]
Thanks, @Dacit. I will have a look at it before the end of the week.

**Review by tancnle — COMMENTED** [review]
Thank you for your hard work @Dacit 🙇🏼 I have left some suggestions for your consideration. Please let me know what you think 🙏🏼 

Could you also kindly add an entry to the [Languages](https://github.com/rouge-ruby/rouge/blob/master/docs/Languages.md) file?

File: lib/rouge/lexers/isabelle.rb [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        rule word(keyword_theory_script), Keyword::Namespace
+        rule word(keyword_abandon_proof), Generic::Error
+        rule word(keyword_qed), Keyword
+        rule word(keyword_proof_goal), Keyword
+        rule word(keyword_proof_block), Keyword
+        rule word(keyword_proof_decl), Keyword
+        rule word(keyword_proof_chain), Keyword
+        rule word(keyword_proof_asm), Keyword
+        rule word(keyword_proof_asm_goal), Keyword
+        rule word(keyword_proof_script), Keyword::Pseudo
```
[the remark this answers is not in the document]
Comment (Dacit) [reply lib/rouge/lexers/isabelle.rb]: I checked that in detail and yes it is possible.

**Comment by Dacit** [comment]
@tancnle Thanks for your suggestions, I integrated all of them. Ready to merge now?

**Comment by tancnle** [comment]
Thank you for considering the suggestions 🙇🏼 Looks good to me 👍🏼 🚀

**Pull request merged**